src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,34 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    public static void main(String[] args) {
+        FileReader fileReader = new FileReaderImpl();
+        List<String> inputReport = fileReader.read("reportToRead.csv");
+
+        DataConverter dataConverter = new DataConverterImpl();
+        final List<FruitTransaction> transactions = dataConverter.convertToTransaction(inputReport);
+
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlers = new HashMap<>();
+        Map<String, Integer> fruitStorage = new HashMap<>();
+        operationHandlers.put(FruitTransaction.Operation.BALANCE, 
+                new BalanceOperation(fruitStorage));
+        operationHandlers.put(FruitTransaction.Operation.PURCHASE, 
+                new PurchaseOperation(fruitStorage));
+        operationHandlers.put(FruitTransaction.Operation.RETURN, 
+                new ReturnOperation(fruitStorage));
+        operationHandlers.put(FruitTransaction.Operation.SUPPLY, 
+                new SupplyOperation(fruitStorage));
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlers);
+
+        ShopService shopService = new ShopServiceImpl(operationStrategy);
+        shopService.process(transactions);
+
+        ReportGenerator reportGenerator = new ReportGeneratorImpl(fruitStorage);
+        String resultingReport = reportGenerator.getReport();
+
+        FileWriter fileWriter = new FileWriterImpl();
+        fileWriter.write(resultingReport, "finalReport.csv");
+    }
+}
```

Comment:
The Main class should be placed in a separate package, not directly under the 'core.basesyntax'. Consider creating a package like 'core.basesyntax.app' for it.